packages/did/src/Did.chain.ts

```diff
@@ -106,248 +144,277 @@ export async function queryAllServicesEncoded(
-export async function queryEndpointsCountsEncoded(
-  didIdentifier: IIdentity['address']
+async function queryEndpointsCountsEncoded(
+  didIdentifier: IDidIdentifier
 ): Promise<u32> {
   const { api } = await BlockchainApiConnection.getConnectionOrConnect()
   return api.query.did.didEndpointsCount<u32>(didIdentifier)
 }
 
-async function queryDepositAmountEncoded(): Promise<U128> {
+async function queryDepositAmountEncoded(): Promise<u128> {
   const { api } = await BlockchainApiConnection.getConnectionOrConnect()
-  return api.consts.did.deposit as U128
+  return api.consts.did.deposit as u128
+}
+
+// ### DECODED QUERYING types
+
+export type IChainDeposit = {
+  owner: IIdentity['address']
+  amount: BN
+}
+
+export type IDidChainRecordJSON = {
+  authenticationKey: DidKey['id']
+  keyAgreementKeys: Array<DidKey['id']>
+  capabilityDelegationKey?: DidKey['id']
+  assertionMethodKey?: DidKey['id']
+  publicKeys: DidKey[]
+  lastTxCounter: BN
+  deposit: IChainDeposit
 }
 
 // ### DECODED QUERYING (builds on top of raw querying)
 
-// This should not be part of this layer, as it has knowledge of DID URI.
-// This level should only be concerned with IDs.
-// Building DID URIs from IDs should be a concern of a higher level, so
-// we might want to refactor this in the future when time pressure is off.
-function assembleKeyId(keyId: Codec, did: IDidDetails['did']): string {
-  return `${did}#${keyId.toHex()}`
+function decodeDidDeposit(encodedDeposit: Deposit): IChainDeposit {
+  return {
+    amount: new BN(encodedDeposit.amount.toString()),
+    owner: encodedDeposit.owner.toString(),
+  }
 }
 
 function decodeDidPublicKeyDetails(
-  did: IDidDetails['did'],
   keyId: Hash,
   keyDetails: DidPublicKeyDetails
-): IDidKeyDetails {
+): DidKey {
   const key = keyDetails.key.value
   return {
-    id: assembleKeyId(keyId, did),
+    id: keyId.toHex(),
     type: key.type.toLowerCase(),
-    controller: did,
-    publicKeyHex: key.value.toHex(),
-    includedAt: keyDetails.blockNumber.toNumber(),
+    publicKey: key.value.toU8a(),
+    includedAt: keyDetails.blockNumber.toBn(),
   }
 }
 
-// Same reasoning as `assembleKeyId`.
 function decodeDidChainRecord(
-  didDetail: IDidChainRecordCodec,
-  did: IDidDetails['did']
+  didDetail: IDidChainRecordCodec
 ): IDidChainRecordJSON {
-  const publicKeys: IDidKeyDetails[] = Array.from(
-    didDetail.publicKeys.entries()
-  ).map(([keyId, keyDetails]) => {
-    return decodeDidPublicKeyDetails(did, keyId, keyDetails)
-  })
-  const authenticationKeyId = assembleKeyId(didDetail.authenticationKey, did)
-  const keyAgreementKeyIds = Array.from(
-    didDetail.keyAgreementKeys.values()
-  ).map((id) => assembleKeyId(id, did))
+  const publicKeys: DidKey[] = [...didDetail.publicKeys.entries()].map(
+    ([keyId, keyDetails]) => {
+      return decodeDidPublicKeyDetails(keyId, keyDetails)
+    }
+  )
+  const authenticationKeyId = didDetail.authenticationKey.toHex()
```

Comment:
Authentication key can be rotated, but it is the only key that is required to always be present. Hence, it can never be undefined, otherwise the DID would become unusable.